src/list.rs

```diff
@@ -32,8 +33,47 @@ fn handle_list(app_state: &mut AppState, stdout: &mut StdoutLock) -> Result<()>
 
                 list_state.message.clear();
 
+                let curr_key = key.code;
+
+                if is_searching {
+                    match curr_key {
+                        KeyCode::Esc | KeyCode::Enter => {
+                            is_searching = false;
+                            list_state.search_query.clear();
+                            continue;
```

Comment:
We continue in each match arm. It would be better to use `continue` at the end of the `if` body instead.